### DISCUSSION TITLE
RFC : Dynamic Question-Set.

### DISCUSSION THREAD

**Discussion by mahesh-maney** [Sunbird-inQuiry/Community#50] [Feature Requests]
Requirement

For UPSMF (Uttar Pradesh State Medical Faculty), As a User, system should enable user to automate pulling of questions into question-set based on taxonomy association. 
:map: Background

In UPSMF System, Most of the questions are sourced through a question bank. Users with creation access to the system are expected to create Blueprints. Blueprints are form of question Set which do not have questions explicitly marked against question set, rather have a criteria to pull these questions from question bank.

In order to enable automatic pulling of questions against a blue print. There is a need to build an extension to the system to enable the existing system to explicitly automate the question pulling into the Question-Set(s).

A detailed document is available here -> https://project-sunbird.atlassian.net/wiki/spaces/UPSMF/pages/3261890567/Dynamic+Question-Set.

**Comment by Krishnaj20** [comment]
tagging @vrayulu @vinukumar-vs @pallakartheekreddy @coolbung @aks30 @pkranga @vijiurs for feedback

**Comment by Krishnaj20** [comment]
@mahesh-maney - QR is synonymous with QR codes in Sunbird context hence to avoid confusion please re-name it appropriately - for ex., question contributor. Can you also call out the reporting needs and how you are planning to address them please?

@vrayulu - Please review and share your feedback.

**Comment by aks30** [comment]
This can be an exciting feature, I see potentially this being used in preparing practice question sets, etc. Though cannot be used for where there has to be a result/comparison to be made at the end because each user might have a different test completely.

**Comment by abhay-samagra** [comment]
@kameshbhr - Seems like there is a ticket here already, closing the [one](https://github.com/orgs/Sunbird-coKreat/discussions/90) in coKreat, and reviving this thread.

**Comment by kameshbhr** [comment]
@mahesh-maney @PoojaJManjooran @sajeshkayyath What is the status of this RFC? I see some open questions in the attached confluence document.  I too have few questions on the functionality and tech design.

**Reply by PoojaJManjooran** [reply]
Once the open questions are addressed, we should be having a DC review for finalising the approach I guess

**Reply by abhay-samagra** [reply]
@mahesh-maney - Please update the documentation to close the open comments. That seems to be a blocker for having the DC review.

**Reply by mahesh-maney** [reply]
@abhay-samagra , @PoojaJManjooran , @sajeshkayyath , @kameshbhr I have replied to your comments on the RFC wiki page. Please check.
